pkg/controller/util/endpoint/controller_utils.go

```diff
@@ -135,9 +135,15 @@ func DeepHashObjectToString(objectToWrite interface{}) string {
 	return hex.EncodeToString(hasher.Sum(nil)[0:])
 }
 
-// ShouldPodBeInEndpointSlice returns true if a specified pod should be in an EndpointSlice object.
+// ShouldPodBeInEndpointSlice returns true if a specified pod should be in an Endpoint or EndpointSlice object.
 // Terminating pods are only included if includeTerminating is true
 func ShouldPodBeInEndpointSlice(pod *v1.Pod, includeTerminating bool) bool {
```

Comment:
Maybe `Endpoints` is a better term here due to being shorter and more clearly applying to both Endpoints and EndpointSlices (EndpointSlices contain an "Endpoints" list)
```suggestion
func ShouldPodBeInEndpoints(pod *v1.Pod, includeTerminating bool) bool {
```